pkg/handlers/v1/cloud_fetch.go

```diff
@@ -0,0 +1,145 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/asecurityteam/asset-inventory-api/pkg/domain"
+	"github.com/asecurityteam/asset-inventory-api/pkg/logs"
+)
+
+// CloudAssets represents a list of assets
+type CloudAssets struct {
+	Assets []CloudAssetDetails `json:"assets"`
+}
+
+// CloudAssetDetails represent an asset and associated attributes
+type CloudAssetDetails struct {
+	PrivateIPAddresses []string          `json:"privateIpAddresses"`
+	PublicIPAddresses  []string          `json:"publicIpAddresses"`
+	Hostnames          []string          `json:"hostnames"`
+	ResourceType       string            `json:"resourceType"`
+	AccountID          string            `json:"accountId"`
+	Region             string            `json:"region"`
+	ResourceID         string            `json:"resourceId"`
+	Tags               map[string]string `json:"tags"`
+}
+
+// CloudAssetFetchByIPParameters represents the incoming payload for fetching cloud assets by IP address
+type CloudAssetFetchByIPParameters struct {
+	IPAddress string `json:"ipAddress"`
+	Timestamp string `json:"time"`
+}
+
+// CloudAssetFetchByHostnameParameters represents the incoming payload for fetching cloud assets by hostname
+type CloudAssetFetchByHostnameParameters struct {
+	Hostname  string `json:"hostname"`
+	Timestamp string `json:"time"`
+}
+
+// CloudFetchByIPHandler defines a lambda handler for fetching cloud assets with a given IP address
+type CloudFetchByIPHandler struct {
+	LogFn   domain.LogFn
+	StatFn  domain.StatFn
+	Fetcher domain.CloudAssetByIPFetcher
+}
+
+// Handle handles fetching cloud assets by IP address
+func (h *CloudFetchByIPHandler) Handle(ctx context.Context, input CloudAssetFetchByIPParameters) (CloudAssets, error) {
+	logger := h.LogFn(ctx)
+
+	ts, e := time.Parse(time.RFC3339Nano, input.Timestamp)
+	if e != nil {
+		logger.Info(logs.InvalidInput{Reason: e.Error()})
+		return CloudAssets{}, InvalidInput{Field: "time", Cause: e}
+	}
+
+	if input.IPAddress == "" {
```

Comment:
Assuming input validation fails or is disabled, is there any harm in making a query for `ipAddress == ""`? I'm not sure if this is actually a bad request or a not found condition.

Reply:
I think that would depend on how the underlying storage mechanism handles a query for an empty string. It'd be putting unnecessary load on the database - the real question would be how much.

That said, this should be rejected by input validation though, so I think it makes sense to remove it.